lib/rubygems/commands/install_command.rb

```diff
@@ -196,31 +196,19 @@ def install_gem(name, version) # :nodoc:
 
     dinst = Gem::DependencyInstaller.new options
 
-    if options[:ignore_dependencies]
-      install_gem_without_dependencies dinst, name, req
-    else
-      install_gem_with_dependencies dinst, name, req
-    end
-  end
+    request_set = dinst.resolve_dependencies name, req
 
-  def install_gem_without_dependencies(dinst, name, req) # :nodoc:
-    gem = nil
+    if options[:explain]
+      say "Gems to install:"
 
-    if local?
```

Comment:
why are we removing this check?

Reply:
I made several checks to conclude that we can remove it:

* Removing it doesn't break any tests.
* The `:domain` option is passed and theorically handled by the request set anyways.
* This code path is currently broken anyways, and has been for a long time. In both master, 3.0.2, 2.7.8, and 2.5.2, running `gem install rails --local --ignore-dependencies` crashes with 

```
ERROR:  While executing gem ... (TypeError)
    no implicit conversion of nil into String
```

After this PR it no longer crashes but it gives a better error instead.

Reply:
Ok, I'm now learning how you use the `--local` option. You need to specify a locally built gem with it. In that case, after this PR it seems to be still working as before. In any case, let's hold this PR and I'll first refactor this on a separate PR, adding test coverage for the refactor so we can double check we're good here.

```
```

Reply:
I opened #2650 to address this. Once that's merged, I'll rebase this PR on top of it.

Reply:
Marking this as solved since I justified the removal in #2650.